sources/ghostdog/src/main.rs

```diff
@@ -476,4 +501,77 @@ mod test {
             }
         }
     }
+
+    #[test]
+    fn test_write_config_string_success() -> Result<()> {
+        // Create a temporary directory for testing
+        let temp_dir = TempDir::new().unwrap();
+        let config_path = temp_dir.path().join("config.txt");
+        let test_content = "test content";
+
+        // Write content to file
+        write_config_string(config_path.clone(), test_content)?;
+
+        // Verify the content was written correctly
+        let read_content = fs::read_to_string(&config_path).unwrap();
+        assert_eq!(read_content, format!("{}\n", test_content));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_write_config_string_no_parent_directory() {
+        // Try to write to a path without a parent directory
+        let result = write_config_string(PathBuf::from("file.txt"), "content");
+        println!("{:?}", result);
+        assert!(matches!(
+            result.unwrap_err(),
+            error::Error::NoParentDirectory { path: _ }
+        ));
+        // Try to write to /
+        let result = write_config_string(PathBuf::from("/file.txt"), "content");
+
+        assert!(matches!(
+            result.unwrap_err(),
+            error::Error::NoParentDirectory { path: _ }
+        ));
+    }
+
+    #[test]
+    fn test_write_config_string_empty_content() -> Result<()> {
+        // Create a temporary directory for testing
+        let temp_dir = TempDir::new().unwrap();
+        let config_path = temp_dir.path().join("config.txt");
+        let empty_content = "";
+
+        // Write empty content to file
+        write_config_string(config_path.clone(), empty_content)?;
+
+        // Verify the content was written correctly
+        let read_content = fs::read_to_string(&config_path).unwrap();
+        assert_eq!(read_content, ""); // Note: writeln! adds a newline
```

Comment:
nit: can move this comment to the next test:
```suggestion
        assert_eq!(read_content, "");
```